### ISSUE TITLE
Pydantic ImportString for oauth_cb in ConnectionConfig

### ISSUE THREAD

**Issue by mkp-jansen** [quixai/quix-streams#679]
**Is your feature request related to a problem? Please describe.**
We would like to configure our Quix applications with OAuth authentication through YAML files. For example:

```YAML
quix-app:
  broker_address:
    bootstrap_servers: localhost:9092
    security_protocol: sasl_ssl
    sasl_mechanism: OAUTHBEARER
  consumer_group: quix-consumer
  auto_offset_reset: earliest
```

Currently, this can easily be achieved with something like this:

```python
# in my_module.py
def my_oauth_cb(config: str) -> Tuple[str, float]:
    # get token with config
    return "token", 1.0

with open("config.yml") as f:
    config = yaml.safe_load(f)

# setting oauth_cb in the code
config["quix-app"]["broker_address"]["oauth_cb"] = my_oauth_cb

app = Application(
    **config["quix-app"]
)
```
However, passing the `oauth_cb` function has to be done in the code. It would be beneficial if we could specify the callback via Pydantic's `ImportString` directly from the YAML file. For example:

```YAML
quix-app:
  broker_address:
    ...
    oauth_cb: my_module.my_oauth_cb
    ...
```
Only a minor adjustment in quixstreams/kafka/configuration.py would be necessary - without impacting any existing functionality:

```python
# file: quixstreams/kafka/configuration.py - lines: 55 - 62
oauth_cb: Optional[Union[Callable[[str], Tuple[str, float]], ImportString]] = (
    pydantic.Field(
        # Prevent the AliasGenerator from changing the field name to "oauth.cb"
        default=None,
        alias_priority=2,
        serialization_alias="oauth_cb",
    )
)
```

If it is OK, I would create a PR for the feature.

**Comment by mkp-jansen** [comment]
#680

**Comment by ghost** [comment]
Hey @mkp-jansen !
The change makes sense to me.
I'll review the PR 👍

**Comment by ghost** [comment]
It's merged 👍 
I'll drop a comment here when it's released

**Comment by mkp-jansen** [comment]
thanks, again!

**Comment by ghost** [comment]
Released in [v3.5.0](https://github.com/quixio/quix-streams/releases/tag/v3.5.0)
